### PR TITLE
Fixing triangle.c to allow windows 64

### LINKED ISSUE #5828: PFEM-Fluid 2D fails on Windows [Triangle?]
Hi @AFranci 

I'm unable to run a basic 2D case of PFEMFluid app in Windows.
The process stops after this output:
```

 |  /           |
 ' /   __| _` | __|  _ \   __|
 . \  |   (   | |   (   |\__ \
_|\_\_|  \__,_|\__|\___/ ____/
           Multi-Physics 7.0.0-3efef92328-Release
Compiled with OpenMP support.
Maximum OpenMP threads: 1.
Importing    KratosExternalSolversApplication 
Initializing KratosExternalSolversApplication...
Importing    KratosDelaunayMeshingApplication 
            ___      _                                  
    KRATOS |   \ ___| |__ _ _  _ _ _  __ _ _  _         
           | |) / -_| / _` | || | ' \/ _` | || |        
           |___/\___|_\__,_|\_,_|_||_\__,_|\_, | MESHING
                                            |__/        
Initialize KratosDelaunayMeshingApplication...
Importing    KratosPfemFluidDynamicsApplication 
            ___  __           ___ _      _    _          
     KRATOS| _ \/ _|___ _ __ | __| |_  _(_)__| |         
           |  _/  _/ -_) '  \| _|| | || | / _` |         
           |_| |_| \___|_|_|_|_| |_|\_,_|_\__,_|DYNAMICS 
Initializing KratosPfemFluidDynamicsApplication...       
Importing    KratosConstitutiveModelsApplication 
            __  __      _           _      _          
    KRATOS |  \/  |__ _| |_ ___ _ _(_)__ _| |         
           | |\/| / _` |  _/ -_) '_| / _` | |         
           |_|  |_\__,_|\__\___|_| |_\__,_|_| MODELS
Initialize KratosConstitutiveModelsApplication... 
Importing    KratosSolidMechanicsApplication 
            ___      _ _    _           
    KRATOS / __| ___| (_)__| |          
           \__ \/ _ \ | / _` |          
           |___/\___/_|_\__,_| MECHANICS
Initialize KratosSolidMechanicsApplication...  
Tue Oct 22 16:44:39 2019 
  
::[KPFEM Simulation]:: [OMP USING1THREADS ] 
  
::[KPFEM Simulation]:: [Time Step:0.001 echo:0] 
Linear-Solver-Factory: Constructing a regular (non-complex) linear-solver 
Linear-Solver-Factory: Constructing a regular (non-complex) linear-solver 
::[PythonSolver]::: Reading model part. 
::[PythonSolver]::: Reading model part from file: C:\Users\garat\Desktop\aaa.gid\aaa.mdpa 
ModelPartIO:   [Reading Nodes    : 744 nodes read]
ModelPartIO:   [Reading Elements : 978 elements read] [Type: TwoStepUpdatedLagrangianVPFluidElement2D]
ModelPartIO:   [Reading Elements : 16 elements read] [Type: UpdatedLagrangianVSolidElement2D]
ModelPartIO:   [Reading Elements : 17 elements read] [Type: Element2D2N]
ModelPartIO:   [Reading Elements : 238 elements read] [Type: Element2D2N]
ModelPartIO:   [Total Lines Read : 4371]
::[PythonSolver]::: Finished reading model part from mdpa file. 
ModelPart: -PfemFluidModelPart- model part
    Buffer Size : 1
    Number of tables : 0
    Number of sub model parts : 5
    Current solution step index : 0
    SPACE_DIMENSION : 2
    DOMAIN_SIZE : 2
    DELTA_TIME : 0.001
    TIME : 0
    GRAVITY : [3](0,-9.81,0)

    Mesh 0 :
        Number of Nodes       : 744
        Number of Properties  : 5
        Number of Elements    : 1249
        Number of Conditions  : 0
        Number of Constraints : 0

    -Parts_Fluid- model part
        Number of tables : 0
        Number of sub model parts : 0

        Mesh 0 :
            Number of Nodes       : 534
            Number of Properties  : 0
            Number of Elements    : 978
            Number of Conditions  : 0
            Number of Constraints : 0
    -Parts_Interface- model part
        Number of tables : 0
        Number of sub model parts : 0

        Mesh 0 :
            Number of Nodes       : 18
            Number of Properties  : 0
            Number of Elements    : 17
            Number of Conditions  : 0
            Number of Constraints : 0
    -Parts_Rigid_Walls- model part
        Number of tables : 0
        Number of sub model parts : 0

        Mesh 0 :
            Number of Nodes       : 239
            Number of Properties  : 0
            Number of Elements    : 238
            Number of Conditions  : 0
            Number of Constraints : 0
    -Parts_Solid- model part
        Number of tables : 0
        Number of sub model parts : 0

        Mesh 0 :
            Number of Nodes       : 18
            Number of Properties  : 0
            Number of Elements    : 16
            Number of Conditions  : 0
            Number of Constraints : 0
    -VELOCITY_Rigid_Walls- model part
        Number of tables : 0
        Number of sub model parts : 0

        Mesh 0 :
            Number of Nodes       : 239
            Number of Properties  : 0
            Number of Elements    : 0
            Number of Conditions  : 0
            Number of Constraints : 0
 
::[PythonSolver]:: : Finished reading model part. 
InletManagementProcess:  inlet_management CONSTRUCTOR RemoveMeshNodesForFluidsProcess:  remove_mesh_nodes_process_for_fluids ComputeAveragePfemMeshParametersProcess:  inlet_management CONSTRUCTOR Model: DEPRECATION_WARNING: The ModelPart "VELOCITY_Rigid_Walls"
is retrieved from the Model by using the flat-map!
This will be removed end of November 2019
Please prepend the Parent-ModelPart-Names like this:
"PfemFluidModelPart.VELOCITY_Rigid_Walls"
Model: DEPRECATION_WARNING: The ModelPart "VELOCITY_Rigid_Walls"
is retrieved from the Model by using the flat-map!
This will be removed end of November 2019
Please prepend the Parent-ModelPart-Names like this:
"PfemFluidModelPart.VELOCITY_Rigid_Walls"
Model: DEPRECATION_WARNING: The ModelPart "VELOCITY_Rigid_Walls"
is retrieved from the Model by using the flat-map!
This will be removed end of November 2019
Please prepend the Parent-ModelPart-Names like this:
"PfemFluidModelPart.VELOCITY_Rigid_Walls"
  
::[KPFEM Simulation]:: Analysis -START-  


```

The main suspect is the Triangle library, who has problems with 2D for Win64
source: @escolano @pooyan-dadvand

### PR DESCRIPTION
**Pull request by jginternational**
With the solution found here:
https://stackoverflow.com/questions/44558793/triangle-library-with-vs2015-x64

Fixes #5828

### CODE REVIEW

File: external_libraries/triangle/triangle.c [hunk 1]
```diff
@@ -635,7 +635,7 @@ struct memorypool {
   int itembytes;
   int itemsperblock;
   int itemsfirstblock;
-  long items, maxitems;
+  __int64 items, maxitems;
```
Comment (loumalouomega) [inline external_libraries/triangle/triangle.c]: I thing We should use std::size_t, otherwise it will not work in Raspberry Pi

Comment (loumalouomega) [reply external_libraries/triangle/triangle.c]: Well, technically std::ptrdiff_t and std::size_t, for unsigned __int64

**Review by loumalouomega — CHANGES_REQUESTED** [review]
Replace __int64 with std::ptrdiff_t and unsigned __int64 with std::size_t

**Comment by AFranci** [comment]
Thanks Javi! This sounds great!! 
@AlejandroCornejo @maceligueta can you please check it in your pc too?

**Review by AFranci — DISMISSED** [review]

**Comment by philbucher** [comment]
since this is a systematic thing I suggest using a typedef, this way can be changed easier

**Comment by AlejandroCornejo** [comment]
Oh!!! that's very good news!!!!!

**Comment by ipouplana** [comment]
> Oh!!! that's very good news!!!!!

Indeed!!

**Review by loumalouomega — DISMISSED** [review]
OK

**Review by loumalouomega — APPROVED** [review]

**Pull request merged**